python/src/aiconfig/schema.py

```diff
@@ -444,30 +445,134 @@ def get_model_metadata(
     # `update_model_parser_registry_with_config_runtime`` function
     # Tracked in https://github.com/lastmile-ai/aiconfig/issues/503
     def update_model(
-        self, model_metadata: Dict | ModelMetadata, prompt_name: Optional[str] = None
+        self, name: Optional[str] = None, settings: Optional[InferenceSettings] = None, prompt_name: Optional[str] = None
     ):
         """
-        Updates model settings in AIconfig-level metadata
+        Updates model name and/or settings at the prompt (if specified) or AIConfig level.
 
         Args:
-            model_metadata (dict): The model metadata to update.
-            prompt_name (str, optional): If specified, the model settings will only be applied to the prompt with the given prompt_name.
+            name (str): The model name to update. If None, keep existing name.
+            settings (dict): The model settings to update. If None, keep existing settings.
+            prompt_name (Optional[str]): If specified, the model updatd will only be applied to the prompt with the given prompt_name.
         """
-        if isinstance(model_metadata, dict):
-            if "name" not in model_metadata:
-                raise KeyError(
-                    "Cannot update model. Model metadata must contain a 'name' element. Optionally, it may contain a 'settings' element."
-                )
-            model_metadata = ModelMetadata(**model_metadata)
-        if prompt_name:
+        if name is None and settings is None:
+            raise ValueError(
+                "Cannot update model. Either model name or model settings must be specified."
+            )
+        if name is None and prompt_name is None: #Only settings param is set
+            raise ValueError(
+                """
+Cannot update model. There are two things you are trying: \
+    1) Update the settings of a prompt \
+        Fix: You must pass in a `prompt_name` argument \
+    2) Update the settings at the AIConfig-level \
+        Fix: You must pass in a `name` for the model you wish \
+        to update. AIConfig-level can have multiple models, \
+        so without a model name, we don't know which model \
+        to set the settings for."
+"""
+            )
+        
+        # We first update the model name, then update the model settings
+        if name is not None:
+            # We need to pass in settings param because it's needed for
+            # config-level updates
+            self._update_model_name(name, settings, prompt_name)
+       
+        if settings is not None and prompt_name is not None:
+            self._update_model_settings_for_prompt(settings, prompt_name)
+    
+    def _update_model_name(self, name: str, settings: Union[InferenceSettings, None], prompt_name: Optional[str] = None):
```

Comment:
Does this method need to take in settings? since the method name is `update_model_name` It sounds like it should just be operating on the name

Reply:
I left comments in the docstrings. We need the `settings` param to update at ai-config-level